### PR TITLE
Regenerate E2E test logs for release 0.8.0

### PR DESCRIPTION
**Pull request by ckadner**
**Description of your changes:**

Regenerate end-to-end test logs for release v0.8.0

**Environment tested:**

* Python Version (use `python --version`): 3.9
* Tekton Version (use `tkn version`): 0.21.0
* Kubernetes Version (use `kubectl version`): 1.18
* OS (e.g. from `/etc/os-release`): MacOS Cataline

/cc @Tomcli

### CODE REVIEW

**Comment by ckadner** [comment]
```Bash
$ git show --pretty="" --name-only -1 | grep ".log" | sort | while read f; do echo -e  "\n=== [${f}](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/${f}) ===\n"; cat $f | grep -iE "error|fail"; done
```

=== [sdk/python/tests/compiler/testdata/any_sequencer.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/any_sequencer.log) ===

[flipcoin : main] 2021/04/23 20:01:14 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/artifact_passing_using_volume.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/artifact_passing_using_volume.log) ===

[processor : copy-artifacts] tar: error exit delayed from previous errors

=== [sdk/python/tests/compiler/testdata/basic_no_decorator.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/basic_no_decorator.log) ===

[get-frequent : main] 2021/04/23 20:08:59 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/big_data_passing.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/big_data_passing.log) ===

[gen-params : copy-artifacts] tar: error exit delayed from previous errors

=== [sdk/python/tests/compiler/testdata/cache.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/cache.log) ===

[cache-enabled : main] 2021/04/23 20:12:04 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/compose.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/compose.log) ===


=== [sdk/python/tests/compiler/testdata/condition.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/condition.log) ===

[flip : main] 2021/04/23 20:15:02 Error executing command: fork/exec /bin/bash: no such file or directory
[flip3 : main] 2021/04/23 20:15:03 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/condition_custom_task.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/condition_custom_task.log) ===

[flip-coin : main] 2021/04/23 20:15:25 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/condition_dependency.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/condition_dependency.log) ===

[flip : main] 2021/04/23 20:15:47 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/conditions_and_loops.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/conditions_and_loops.log) ===

[produce-numbers : main] 2021/04/23 20:16:04 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/conditions_with_global_params.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/conditions_with_global_params.log) ===

[add-numbers : main] 2021/04/23 20:16:21 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/create_component_from_func.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/create_component_from_func.log) ===


=== [sdk/python/tests/compiler/testdata/exit_handler.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/exit_handler.log) ===

[gcs-download : main] 2021/04/23 20:19:12 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/hidden_output_file.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/hidden_output_file.log) ===

[download-file : main] 2021/04/23 20:19:39 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/imagepullsecrets.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/imagepullsecrets.log) ===

[get-frequent : main] 2021/04/23 20:19:56 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/load_from_yaml.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/load_from_yaml.log) ===

[busybox : copy-artifacts] tar: error exit delayed from previous errors

=== [sdk/python/tests/compiler/testdata/long_param_name.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/long_param_name.log) ===

[gcs-downloadgcs-downloadgcs-downloadgcs-download : main] 2021/04/23 20:21:34 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/long_pipeline_name.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/long_pipeline_name.log) ===


=== [sdk/python/tests/compiler/testdata/loop_over_lightweight_output.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/loop_over_lightweight_output.log) ===

[produce-list : main] 2021/04/23 20:22:14 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/loop_static.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/loop_static.log) ===


=== [sdk/python/tests/compiler/testdata/opsgroups.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/opsgroups.log) ===


=== [sdk/python/tests/compiler/testdata/parallel_join.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/parallel_join.log) ===

[gcs-download : main] 2021/04/23 20:23:44 Error executing command: fork/exec /bin/bash: no such file or directory
[gcs-download-2 : main] 2021/04/23 20:23:44 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/parallel_join_with_argo_vars.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/parallel_join_with_argo_vars.log) ===

[gcs-download : main] 2021/04/23 20:24:06 Error executing command: fork/exec /bin/bash: no such file or directory
[gcs-download-2 : main] 2021/04/23 20:24:06 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/parallelfor_item_argument_resolving.log) ===

[produce-str : main] 2021/04/23 20:24:28 Error executing command: fork/exec /bin/bash: no such file or directory
[produce-list-of-strings : main] 2021/04/23 20:24:29 Error executing command: fork/exec /bin/bash: no such file or directory
[produce-list-of-ints : main] 2021/04/23 20:24:29 Error executing command: fork/exec /bin/bash: no such file or directory
[produce-list-of-dicts : main] 2021/04/23 20:24:29 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/pipelineparams.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/pipelineparams.log) ===

[download : main] 2021/04/23 20:25:16 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/recur_cond.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/recur_cond.log) ===


=== [sdk/python/tests/compiler/testdata/recursion_while.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/recursion_while.log) ===

[flip-coin : main] 2021/04/23 20:25:55 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/resourceop_basic.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/resourceop_basic.log) ===

[test-step : main] 2021/04/23 20:26:18 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/sidecar.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/sidecar.log) ===

[download : main] 2021/04/23 20:27:33 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/tekton_custom_task.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/tekton_custom_task.log) ===

[flip-coin : main] 2021/04/23 20:27:50 Error executing command: fork/exec /bin/bash: no such file or directory
[flip-coin-2 : main] 2021/04/23 20:27:51 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/tekton_pipeline_conf.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/tekton_pipeline_conf.log) ===


=== [sdk/python/tests/compiler/testdata/tekton_pipeline_variables.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/tekton_pipeline_variables.log) ===


=== [sdk/python/tests/compiler/testdata/volume_op.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/volume_op.log) ===

[create-pvc : main] time="2021-04-23T20:28:56Z" level=error msg="Initialize script failed: exit status 2:"

=== [sdk/python/tests/compiler/testdata/withitem_multi_nested.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/withitem_multi_nested.log) ===


=== [sdk/python/tests/compiler/testdata/withitem_nested.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/withitem_nested.log) ===


=== [sdk/python/tests/compiler/testdata/withparam_global.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/withparam_global.log) ===

[my-out-cop0 : main] 2021/04/23 20:31:54 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/withparam_global_dict.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/withparam_global_dict.log) ===

[my-out-cop0 : main] 2021/04/23 20:32:18 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/withparam_output.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/withparam_output.log) ===

[my-out-cop0 : main] 2021/04/23 20:32:40 Error executing command: fork/exec /bin/bash: no such file or directory

=== [sdk/python/tests/compiler/testdata/withparam_output_dict.log](https://github.com/kubeflow/kfp-tekton/blob/b34358f/sdk/python/tests/compiler/testdata/withparam_output_dict.log) ===

[my-out-cop0 : main] 2021/04/23 20:32:58 Error executing command: fork/exec /bin/bash: no such file or directory

**Comment by Tomcli** [comment]
@ckadner these pipelines we don't have to test e2e
long_param_name.py - unit test, not using real container image.
recur_cond.py - unit test, not using real container image.

**Comment by Tomcli** [comment]
Also there's few example using the experimental CEL custom task. You have to install them separately using golang ko
```
git clone https://github.com/tektoncd/experimental/
cd experimental/cel
ko apply -f config/
```

**Comment by ckadner** [comment]
Thanks @Tomcli -- #567 appears to address the `Error executing command: fork/exec /bin/bash: no such file or directory`

**Comment by ckadner** [comment]
### E2E tests succeeding with errors being logged

- `katib.log`:
  ```
  3:  [mnist-hpo : main] ERROR:root:Exception when create kubeflow.org/experiments: 404 page not found
  17: [mnist-hpo : main]     raise ValueError(errmsg("Extra data", s, end, len(s)))
  18: [mnist-hpo : main] ValueError: Extra data: line 1 column 5 - line 2 column 1 (char 4 - 19)
  ```
- `volume_op.log`:
  ```
  1: [create-pvc : main] time="2021-04-23T20:28:56Z" level=error msg="Initialize script failed: exit status 2:"
  ```
- `artifact_passing_using_volume.log`:
  ```
  20: [processor : copy-artifacts] tar: error exit delayed from previous errors
  ```
- `load_from_yaml.log`:
  ```
  5: [busybox : copy-artifacts] tar: error exit delayed from previous errors
  ```

### Repeatedly failing end-to-end tests

- `cond_recur.yaml`
  ```
  Failed to create a workflow for (condition-and-recur-7371d): admission webhook\\"validation.webhook.pipeline.tekton.dev\\" denied the request: validation failed: non-existent variable in\\"$(params.condition-cel-status)\\": spec.pipelinespec.tasks[1].params[condition-cel-status]"
  ```
- `condition_custom_task.yaml`
  ```
  Timeout after 8 minutes: Status: 'Running', expected: ['Succeeded', 'Completed']
  ```
- `loop_static.yaml`
  ```
  Status: Failed
  ```
- `recursion_while.yaml`
  ```
  Timeout after 8 minutes: Status: 'Running', expected: ['Succeeded', 'Completed']
  ```
- `tekton_custom_task.yaml`
  ```
  Timeout after 8 minutes: Status: 'Running', expected: ['Succeeded', 'Completed']
  ```
- `withitem_multi_nested.yaml`
  ```
  Status: Failed
  ```
- `withitem_nested.yaml`
  ```
  Timeout after 8 minutes: Status: 'Running', expected: ['Succeeded', 'Completed']
  ```
- `withparam_output.yaml`
  ```
  Timeout after 8 minutes: Status: 'Running', expected: ['Succeeded', 'Completed']
  ```

### K8s Pod errors

```
tekton-cel-run     cel-controller-8457cdc8f-jgjk9                                    0/1     ImagePullBackOff   0          44h
t
```
```
$ kubectl logs -n tekton-cel-run     cel-controller-8457cdc8f-jgjk9

Error from server (BadRequest): container "cel-controller" in pod "cel-controller-8457cdc8f-jgjk9" is waiting to start: trying and failing to pull image
```

**Comment by Tomcli** [comment]
@ckadner 
pipeline not for runtime
katib.log - (compiler test for katib variable, we don't maintain it to be running)
cond_recur.yaml - (unit test for recursion under condition group, not a valid running example)

**Comment by Tomcli** [comment]
@ckadner everything under `Repeatedly failing end-to-end tests` is for custom task. The loop custom task needs to be pre-defined before running the pipeline. Is it possible to do something like this?
https://github.com/kubeflow/kfp-tekton/tree/master/samples/nested-loops#instructions

For artifact_passing, load_from_yaml, and volume_op, I can run it on my cluster, but it takes too long because these pipelines need to provision volume dynamically. 

Katib and cond_recur is just unit test, we don't need to run them in e2e.

**Comment by ckadner** [comment]
@Tomcli -- I applied the loop CRDs and the CEL controller. I am down to two failing E2E test cases:

```Bash
$ tkn pipelinerun -n kubeflow list | grep Failed

tekton-custom-task-on-kubeflow-pipeline-c4b79                     7 minutes ago   6 seconds    Failed
conditional-execution-pipeline-e54e0                              9 minutes ago   5 seconds    Failed


$ tkn pipelinerun -n kubeflow logs tekton-custom-task-on-kubeflow-pipeline-c4b79

[flip-coin : main] heads
[flip-coin-2 : main] heads
Tasks Completed: 3 (Failed: 1, Cancelled 0), Skipped: 1


$ tkn pipelinerun -n kubeflow logs conditional-execution-pipeline-e54e0

[flip-coin : main] heads
Tasks Completed: 2 (Failed: 1, Cancelled 0), Skipped: 8
```

On the Kubeflow Pipelines dashboard, when I click on these Runs, I get these respective errors:

`Error: failed to retrieve run: c4b79883-1ca6-4181-bda4-24f80616a6de. Click Details for more information.`
`Error: failed to retrieve run: e54e027c-ce3f-4574-b783-1a943c225efe. Click Details for more information.`

**Comment by ckadner** [comment]
[Enabling auto-strip of EOF newlines](https://github.com/kubeflow/kfp-tekton/blob/master/guides/kfp-admin-guide.md#enable-auto-strip-for-end-of-file-newlines) fixed the two failing test cases `condition_custom_task` and 
`tekton_custom_task`:
```
kubectl patch cm kfp-tekton-config -n kubeflow -p '{"data":{"strip_eof":"true"}}'
kubectl rollout restart deploy/ml-pipeline -n kubeflow
```

However, this introduces new errors in other test cases like

- `artifact-passing-pipeline`:
  ```
  [processor : copy-artifacts] awkNF: /producer-Output-2: No such file or directory
  ```
- `load_from_yaml`:
  ```
  [busybox : copy-artifacts] awkNF: /tekton/results/dummy-output-path: No such file or directory
  ```

File: manifests/kustomize/base/pipeline/kfp-pipeline-config.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ data:
         push_artifact main-log step-main.log
     }
     strip_eof() {
+        if [ -f "$2" ]; then
         awk 'NF' $2 | head -c -1 > $1_temp_save && cp $1_temp_save $2
```
Comment (Tomcli) [inline manifests/kustomize/base/pipeline/kfp-pipeline-config.yaml:26]: do we need to indent this line?

**Comment by Tomcli** [comment]
The only thing that concerns me is the size for `big_data_passing.log`. This example is intended to test big log file which it could be too big for regular users to clone this repo.

**Comment by ckadner** [comment]
> The only thing that concerns me is the size for `big_data_passing.log`. This example is intended to test big log file which it could be too big for regular users to clone this repo.

Good point, 2.9 Mb is a large log file. However considering the kfp-tekton repo is already at 35 Mb. I don't think another 3 Mb would make the repo  much less cloneable. 

```Bash
$ du -hs * | sort -rh | head -10

8 M	frontend
5 M	backend
4 M	third_party
4 M	components
3 M	manifests
1 M	samples
1 M	sdk
1 M	guides
1 M	v2
```

Unless you strongly object, I would want to include the big data passing in the end-to-end tests.

File: manifests/kustomize/base/pipeline/kfp-pipeline-config.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ data:
         push_artifact main-log step-main.log
     }
     strip_eof() {
-        awk 'NF' $2 | head -c -1 > $1_temp_save && cp $1_temp_save $2
+        if [ -f "$2" ]; then
+            awk 'NF' $2 | head -c -1 > $1_temp_save && cp $1_temp_save $2
```
Comment (ckadner) [inline manifests/kustomize/base/pipeline/kfp-pipeline-config.yaml:26]: yup, looks better properly indented 👍🏻

File: manifests/kustomize/base/pipeline/kfp-pipeline-config.yaml (hunk 1 above)
Comment (ckadner) [reply manifests/kustomize/base/pipeline/kfp-pipeline-config.yaml]: yup, I amended my last commit, thanks

**Comment by ckadner** [comment]
To summarize, of the non-excluded pipelines, none are failing any longer. 

There are some error message the log files:

```Bash
$ find sdk/python/tests/compiler/testdata -name "*.log" -exec grep -rHniE --color "error|fail" {} \;

artifact_passing_using_volume.log:18: [processor : copy-artifacts] /tekton/scripts/script-1-b4jcg: line 17: workspaces.producer.path: not found
artifact_passing_using_volume.log:19: [processor : copy-artifacts] tar: /producer-Output-2: No such file or directory
artifact_passing_using_volume.log:20: [processor : copy-artifacts] tar: error exit delayed from previous errors

katib.log:3: [mnist-hpo : main] ERROR:root:Exception when create kubeflow.org/experiments: 404 page not found
katib.log:17:[mnist-hpo : main]     raise ValueError(errmsg("Extra data", s, end, len(s)))
katib.log:18:[mnist-hpo : main] ValueError: Extra data: line 1 column 5 - line 2 column 1 (char 4 - 19)

load_from_yaml.log:5: [busybox : copy-artifacts] tar: error exit delayed from previous errors

volume_op.log:1: [create-pvc : main] time="2021-04-23T20:28:56Z" level=error msg="Initialize script failed: exit status 2:"

```

**Comment by Tomcli** [comment]
> To summarize, of the non-excluded pipelines, none are failing any longer.
> 
> There are some error message the log files:
> 
> ```shell
> $ find sdk/python/tests/compiler/testdata -name "*.log" -exec grep -rHniE --color "error|fail" {} \;
> 
> artifact_passing_using_volume.log:18: [processor : copy-artifacts] /tekton/scripts/script-1-b4jcg: line 17: workspaces.producer.path: not found
> artifact_passing_using_volume.log:19: [processor : copy-artifacts] tar: /producer-Output-2: No such file or directory
> artifact_passing_using_volume.log:20: [processor : copy-artifacts] tar: error exit delayed from previous errors
> 
> katib.log:3: [mnist-hpo : main] ERROR:root:Exception when create kubeflow.org/experiments: 404 page not found
> katib.log:17:[mnist-hpo : main]     raise ValueError(errmsg("Extra data", s, end, len(s)))
> katib.log:18:[mnist-hpo : main] ValueError: Extra data: line 1 column 5 - line 2 column 1 (char 4 - 19)
> 
> load_from_yaml.log:4: [busybox : copy-artifacts] tar: /tekton/results/dummy-output-path: No such file or directory
> load_from_yaml.log:5: [busybox : copy-artifacts] tar: error exit delayed from previous errors
> 
> volume_op.log:1: [create-pvc : main] time="2021-04-23T20:28:56Z" level=error msg="Initialize script failed: exit status 2:"
> ```

I added a PR to enhance some of these error messages. These 4 examples are showing expected behavior but just the error messages are not clear.

artifact_passing_using_volume: Artifacts passing via volume are not trackable from UX.
katib: it was using the old alpha api, but here we are only testing can the compiler handle katib variable. We are not trying to run this pipeline
load_from_yaml: This example is meant to have missing artifacts to test the pipeline won't fail because of missing/wrong artifact path.
volume_op: The error message is no longer relevant. I added a PR to point to the new image.

**Comment by ckadner** [comment]
Thanks @Tomcli

**Pull request merged**